common/scala/src/main/scala/whisk/core/containerpool/logging/ElasticSearchLogStore.scala

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.core.containerpool.logging
+
+import java.nio.file.{Path, Paths}
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.HttpRequest
+import akka.http.scaladsl.model.Uri
+import akka.stream.scaladsl.Flow
+import akka.http.scaladsl.model.HttpResponse
+import akka.http.scaladsl.model._
+
+import whisk.common.AkkaLogging
+import whisk.core.entity.{ActivationLogs, Identity, WhiskActivation}
+import whisk.core.containerpool.logging.ElasticSearchJsonProtocol._
+import whisk.core.ConfigKeys
+
+import scala.concurrent.Future
+import scala.concurrent.Promise
+import scala.util.Try
+
+import spray.json._
+
+import pureconfig._
+
+// Schema of logs in ES
+case class UserLogEntry(message: String, tenantId: String, stream: String, time: String, action: String)
+
+object UserLogEntry extends DefaultJsonProtocol {
+  implicit val serdes =
+    jsonFormat(UserLogEntry.apply, "message", "ALCH_TENANT_ID", "stream_str", "time_date", "action_str")
+}
+
+case class ElasticSearchLogStoreConfig(protocol: String,
+                                       host: String,
+                                       port: Int,
+                                       logMessageField: String,
+                                       activationIdField: String,
+                                       requiredHeaders: String)
```

Comment:
Was having trouble here when overriding pureconfig with Ansible. Seems Ansible is not providing the same list format that pureconfig expects. As a result, error occur when `requiredHeaders` type is `List[String`.